File: apps/slf-esport/frontend/src/types/assignment.ts
/**
 * Exercise assignment types
 */

export enum AssignmentStatus {
  PENDING = 'pending',
  IN_PROGRESS = 'in_progress',
  COMPLETED = 'completed',
  SKIPPED = 'skipped',
}

export interface ExerciseAssignment {
  id: number
  player_id: number
  exercise_id: number
  coach_id: number
  title?: string
  description?: string
  target_score?: string
  assigned_date: string
  due_date?: string
  status: AssignmentStatus
  completed_date?: string
  attempts_count: number
  best_score?: string
  player_notes?: string
  coach_feedback?: string
  priority: number
  is_mandatory: boolean
  created_at: string
  updated_at: string
  // Populated by API
  exercise_name?: string
  player_username?: string
  coach_username?: string
}

export interface ExerciseAssignmentCreate {
  exercise_id: number
  player_id: number
  title?: string
  description?: string
  target_score?: string
  due_date?: string
  priority?: number
  is_mandatory?: boolean
}

export interface ExerciseAssignmentUpdate {
  title?: string
  description?: string
  target_score?: string
  due_date?: string
  status?: AssignmentStatus
  priority?: number
  is_mandatory?: boolean
  player_notes?: string
  coach_feedback?: string
  best_score?: string
}

export interface AssignmentStats {
  total_assignments: number
  pending: number
  in_progress: number
  completed: number
  completion_rate: number
  overdue: number
}

// UI constants

export const ASSIGNMENT_STATUS_LABELS: Record<AssignmentStatus, string> = {
  [AssignmentStatus.PENDING]: 'À faire',
  [AssignmentStatus.IN_PROGRESS]: 'En cours',
  [AssignmentStatus.COMPLETED]: 'Complété',
  [AssignmentStatus.SKIPPED]: 'Ignoré',
}

export const ASSIGNMENT_STATUS_COLORS: Record<AssignmentStatus, string> = {
  [AssignmentStatus.PENDING]: 'bg-gray-100 text-gray-800 dark:bg-gray-800 dark:text-gray-300',
  [AssignmentStatus.IN_PROGRESS]: 'bg-blue-100 text-blue-800 dark:bg-blue-900/20 dark:text-blue-300',
  [AssignmentStatus.COMPLETED]: 'bg-green-100 text-green-800 dark:bg-green-900/20 dark:text-green-300',
  [AssignmentStatus.SKIPPED]: 'bg-orange-100 text-orange-800 dark:bg-orange-900/20 dark:text-orange-300',
}

File: apps/slf-esport/backend/app/schemas/assignment.py
"""
Exercise assignment schemas
"""

from datetime import datetime, date
from typing import Optional
from pydantic import BaseModel, Field
from app.models.assignment import AssignmentStatus


# ========== Exercise Assignment Schemas ==========

class ExerciseAssignmentBase(BaseModel):
    """Base exercise assignment schema"""
    title: Optional[str] = None
    description: Optional[str] = None
    target_score: Optional[str] = None
    due_date: Optional[date] = None
    priority: int = Field(default=0, ge=0, le=10)
    is_mandatory: bool = False


class ExerciseAssignmentCreate(ExerciseAssignmentBase):
    """Create exercise assignment"""
    exercise_id: int
    player_id: int  # Who to assign to


class ExerciseAssignmentUpdate(BaseModel):
    """Update exercise assignment"""
    title: Optional[str] = None
    description: Optional[str] = None
    target_score: Optional[str] = None
    due_date: Optional[date] = None
    status: Optional[AssignmentStatus] = None
    priority: Optional[int] = Field(None, ge=0, le=10)
    is_mandatory: Optional[bool] = None
    player_notes: Optional[str] = None
    coach_feedback: Optional[str] = None
    best_score: Optional[str] = None


class ExerciseAssignmentResponse(ExerciseAssignmentBase):
    """Exercise assignment response"""
    id: int
    player_id: int
    exercise_id: int
    coach_id: int
    assigned_date: datetime
    status: AssignmentStatus
    completed_date: Optional[datetime] = None
    attempts_count: int
    best_score: Optional[str] = None
    player_notes: Optional[str] = None
    coach_feedback: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    # Related data (optional, populated by service)
    exercise_name: Optional[str] = None
    player_username: Optional[str] = None
    coach_username: Optional[str] = None

    class Config:
        from_attributes = True


class AssignmentStats(BaseModel):
    """Assignment statistics for a player"""
    total_assignments: int
    pending: int
    in_progress: int
    completed: int
    completion_rate: float  # Percentage
    overdue: int  # Assignments past due date
